src/Analyzer.Core.JsonRuleTests/README.md

```diff
@@ -0,0 +1,46 @@
+# Test Project for Built-in JSON Rules
+This project is designed for quickly adding tests to verify the correctness of built-in JSON rules.  When new rules are added, tests should be added here to make sure the rule is written correctly and help protect against incorrect changes to the rule.
+
+## Test Setup
+A test (or tests) for a given JSON rule consists of 2 parts:
+1. A JSON configuration describing the test(s).
+2. One or more test ARM templates to analyze.  The results of the analysis are compared with the test configuration to assert correctness of the JSON rule.
+
+### JSON Configuration
+To create a set of tests for a JSON rule, a new JSON file is created in the *Tests* directory.  The name of the file must be the same as the `name` property of the JSON rule, with ".json" as the file extension.
+
+For example, a JSON rule like the following:
+``` json
+{
+    "name": "SuperSecuryCheck",
+    "description": "...",
+    "recommendation": "...",
+    "helpUri": "...",
+    "evaluation": {
+      ...
+    }
+}
+```
+... would have tests defined in a configuration file named "Tests/SuperSecuryCheck.json".
+
+The JSON test configuration has the following schema:
+``` json
+[
+    {
+        "Template": "Name of template file to analyze (without file extension).",
+        "ExpectPass": "true/false - Whether the template named in 'Template' is expected to pass for this rule.",
+        "ReportedLines": "Array of integers - If 'ExpectPass' is false, these are the line numbers expected to be reported in the failure."
```

Comment:
Do you mean validating the messages that are returned in a result?

Reply:
I thought at some point we were going to design a way to handle extra information, so if the tool is consumed as a library the user doesn't have to parse the error message, and it's easier to improve how we show it in the CLI

Reply:
I was thinking we might want to do something like:
```
{
	"Template": "Name of template file to analyze (without file extension).",
	"ExpectPass": "true/false - Whether the template named in 'Template' is expected to pass for this rule.",
	"ReportedErrors": [
		{
			"LineNumber": 5
		}
	]
}
```
Or something like that, in case later we want to do something like:

```
{
	"Template": "Name of template file to analyze (without file extension).",
	"ExpectPass": "true/false - Whether the template named in 'Template' is expected to pass for this rule.",
	"ReportedErrors": [
		{
			"LineNumber": 5,
			"VariableName": ""
		},
		{
			"FlaggedString": ""
		}
	]
}
```

Reply:
Writing that example made me think that specifying ExpectPass and an array of lines/errors is kind of unnecessary, an empty list could be interpreted as a test with no errors

Reply:
I like your thinking about this.  At least changing to an array of objects with a LineNumber makes sense.  Then we can add more once we decide how to surface the information.